### PR TITLE
Master edition collection plugin

### PR DESCRIPTION
**Pull request by nhanphan**
- allow specifying max supply, name and uri on the collection
- the fields are informational only and can be freely updated by the creator

### CODE REVIEW

**Review by blockiosaurus — APPROVED** [review]
Looks good, we just need to be very clear about them being informational.

**Comment by MarkSackerberg** [comment]
> Looks good, we just need to be very clear about them being informational.

Since I am creating the docs right now: What exactly do you mean here? That the numbering of the editions has to be handled manually (no actual checks) and that technically there could be more editions than the `maxSupply` suggests?

**Comment by MarkSackerberg** [comment]
What's the reason for having additional name and uri in the plugin? What are we expecting creators to do with it? The collection itself has this already?

File: clients/js/test/plugins/collection/masterEdition.test.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    masterEdition: {
+      authority: {
+        type: 'UpdateAuthority',
+      },
+      maxSupply: 100,
+      name: some('name'),
+      uri: some('uri'),
+    },
+  });
+});
```
Comment (MarkSackerberg) [inline clients/js/test/plugins/collection/masterEdition.test.ts:79]: Aren't those two tests exactly the same?
Were you going for add and update tests here?

Comment (nhanphan) [reply clients/js/test/plugins/collection/masterEdition.test.ts]: good catch, will update

**Comment by nhanphan** [comment]
> > Looks good, we just need to be very clear about them being informational.
> 
> Since I am creating the docs right now: What exactly do you mean here? That the numbering of the editions has to be handled manually (no actual checks) and that technically there could be more editions than the `maxSupply` suggests?

Correct, the plugin itself does not enforce anything and rather is a standard for the ecosystem to have a place to read the data. We have seen asks in the past for many edge cases like "super special extra edition" and specific fixed edition numbers.

**Comment by nhanphan** [comment]
> What's the reason for having additional name and uri in the plugin? What are we expecting creators to do with it? The collection itself has this already?

This is for when the creator needs provenance over for where the edition is "printed" from as opposed to metadata about the collection which may contain other information.

**Pull request merged**